src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int128Converter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert 128-bit signed integer objects to and
+    /// from various other representations.
+    /// </summary>
+    public class Int128Converter : BaseNumberConverter
+    {
+        /// <summary>
+        /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
+        /// </summary>
+        internal override Type TargetType => typeof(Int128);
+
+        /// <summary>
+        /// Convert the given value string to Int128 using the given radix
+        /// </summary>
+        internal override object FromString(string value, int radix)
+        {
+            // We only support decimal and hex radix in the converter.
+            if (radix != 10 && radix != 16)
+            {
+                throw new ArgumentException(SR.Arg_InvalidBase);
+            }
+
+            return value is null ? 0 : Int128.Parse(value, radix == 16 ? NumberStyles.HexNumber : NumberStyles.None);
```

Comment:
The other integer type converters call `Convert.ToIntXXX`, do we not have equivalent `Convert` methods for Int128?

cc @tannergooding

Reply:
No. `Convert` and `IConvertible` aren't very good APIs and are limited by their dependence on `TypeCode` and the latters inability to version. As of right now, API review had decided to not update these because of those limitations.

Directly using `IParsable` for `string to T` conversions and directly using `INumberBase` for `U to T` conversions using `CreateChecked` is likely a better approach moving forward.

We could maybe minimally support `Convert` but there are still weird edge-cases due to `TypeCode` being `object` for these new types and `IConvertible` not supporting conversion to them.

Reply:
We can always revisit this, including for .NET 7, but there needs to be an API proposal for it.

The biggest limitation is really that it simply can't work in all the same scenarios because of how `IConvertible` is designed.

We might be able to do something around `IConvertible` with `DIMs` and a directly calling the `CreateChecked` helper, but it'd require changes to `TypeCode` and that has risks.

Reply:
I logged the issue https://github.com/dotnet/runtime/issues/70104 to track the discussion.